### PR TITLE
Added stubbed tests for Sync Plans via API.

### PR DESCRIPTION
**Pull request by omaciel**
Covers many different test cases for ``Sync Plans``.

### CODE REVIEW

File: tests/foreman/api/test_syncplan.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+
+    @stubbed()
+    @run_only_on('sat')
+    @data(
+        gen_string('alpha', 15),
+        gen_string('alphanumeric', 15),
+        gen_string('numeric', 15),
+        gen_string('latin1', 15),
+        gen_string('utf8', 15),
+        gen_string('html', 15),
```
Comment (JacobCallahan) [inline tests/foreman/api/test_syncplan.py]: any particular reason for 15, or just trying to be explicit without us trying to remove the "10"? :)

File: tests/foreman/api/test_syncplan.py [hunk 2]
```diff
@@ -0,0 +1,399 @@
+"""Unit tests for the ``repositories`` paths."""
+import datetime
```
Comment (JacobCallahan) [inline tests/foreman/api/test_syncplan.py]: we could just import datetime and timedelta from datetime here, if you want to cut down on namespace resolution and import bulk. not even close to a blocker!

**Comment by JacobCallahan** [comment]
ACK

File: tests/foreman/api/test_syncplan.py (hunk 1 above)
Comment (omaciel) [inline tests/foreman/api/test_syncplan.py]: hehehe just following the same pattern that was used to a similar test done via CLI.

File: tests/foreman/api/test_syncplan.py (hunk 2 above)
Comment (omaciel) [inline tests/foreman/api/test_syncplan.py]: Can do :)

**Comment by elyezer** [comment]
ACK

**Comment by elyezer** [comment]
Karma given

**Pull request merged**